include/swift/AST/DiagnosticsSema.def

```diff
@@ -3594,9 +3594,11 @@ ERROR(invalid_nonobjc_extension,none,
 ERROR(objc_in_extension_context,none,
       "members of constrained extensions cannot be declared @objc", ())
 ERROR(objc_in_generic_extension,none,
-      "members of extensions of "
-      "%select{classes from generic context|generic classes}0 "
-      "cannot be declared @objc", (bool))
+      "extensions of generic classes cannot contain "
+      "'@objc' members", ())
+ERROR(objc_in_resilient_extension,none,
+      "extensions of classes built with library evolution support "
+      "cannot contain '@objc' members", ())
```

Comment:
Hmm.  Do we not have an ability to find and register these when the class is constructed?  I guess it would at least need runtime support and so be deployment-gated.

Reply:
This is what the PR is all about :) The new diagnostic is only emitted if LangOpts.EnableObjCResilientClassStubs is not set. In the future the flag will become a deployment target check and the diagnostic will be rephrased.